core/toolbox/toolbox.js

```diff
@@ -0,0 +1,944 @@
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview Toolbox from whence to create blocks.
+ * @author fraser@google.com (Neil Fraser)
+ */
+'use strict';
+
+goog.provide('Blockly.Toolbox');
+
+goog.require('Blockly.Css');
+goog.require('Blockly.Events');
+goog.require('Blockly.Events.Ui');
+goog.require('Blockly.navigation');
+goog.require('Blockly.registry');
+goog.require('Blockly.ToolboxCategory');
+goog.require('Blockly.ToolboxSeparator');
+goog.require('Blockly.Touch');
+goog.require('Blockly.utils');
+goog.require('Blockly.utils.aria');
+goog.require('Blockly.utils.dom');
+goog.require('Blockly.utils.Rect');
+
+goog.requireType('Blockly.CollapsibleToolboxItem');
+goog.requireType('Blockly.IBlocklyActionable');
+goog.requireType('Blockly.IDeleteArea');
+goog.requireType('Blockly.IStyleable');
+goog.requireType('Blockly.IToolbox');
+goog.requireType('Blockly.SelectableToolboxItem');
+goog.requireType('Blockly.ToolboxItem');
+goog.requireType('Blockly.utils.toolbox');
+
+
+/**
+ * Class for a Toolbox.
+ * Creates the toolbox's DOM.
+ * @param {!Blockly.WorkspaceSvg} workspace The workspace in which to create new
+ *     blocks.
+ * @constructor
+ * @implements {Blockly.IBlocklyActionable}
+ * @implements {Blockly.IDeleteArea}
+ * @implements {Blockly.IStyleable}
+ * @implements {Blockly.IToolbox}
+ */
+Blockly.Toolbox = function(workspace) {
+  /**
+   * The workspace this toolbox is on.
+   * @type {!Blockly.WorkspaceSvg}
+   * @protected
+   */
+  this.workspace_ = workspace;
+
+  /**
+   * The JSON describing the contents of this toolbox.
+   * @type {!Blockly.utils.toolbox.ToolboxJson}
+   * @protected
+   */
+  this.toolboxDef_ = workspace.options.languageTree || {'contents': []};
+
+  /**
+   * Whether the toolbox should be laid out horizontally.
+   * @type {boolean}
+   * @private
+   */
+  this.horizontalLayout_ = workspace.options.horizontalLayout;
+
+  /**
+   * The html container for the toolbox.
+   * @type {?Element}
+   */
+  this.HtmlDiv = null;
+
+  /**
+   * The html container for the contents of a toolbox.
+   * @type {?Element}
+   * @protected
+   */
+  this.contentsDiv_ = null;
+
+  /**
+   * The list of items in the toolbox.
+   * @type {!Array<!Blockly.ToolboxItem>}
+   * @protected
+   */
+  this.contents_ = [];
+
+  /**
+   * The width of the toolbox.
+   * @type {number}
+   * @protected
+   */
+  this.width_ = 0;
+
+  /**
+   * The height of the toolbox.
+   * @type {number}
+   * @protected
+   */
+  this.height_ = 0;
+
+  /**
+   * Is RTL vs LTR.
+   * @type {boolean}
+   */
+  this.RTL = workspace.options.RTL;
+
+  /**
+   * The flyout for the toolbox.
+   * @type {?Blockly.Flyout}
+   * @private
+   */
+  this.flyout_ = null;
+
+  /**
+   * A map from toolbox item IDs to toolbox items.
+   * @type {!Object<string, Blockly.ToolboxItem>}
+   * @protected
+   */
+  this.contentIds_ = {};
+
+  /**
+   * Position of the toolbox and flyout relative to the workspace.
+   * @type {number}
```

Comment:
Could do
```
/**
 * ENUM for toolbox and flyout at top of screen.
 * @const
 */
Blockly.TOOLBOX_AT_TOP = 0;

/**
 * ENUM for toolbox and flyout at bottom of screen.
 * @const
 */
Blockly.TOOLBOX_AT_BOTTOM = 1;

/**
 * ENUM for toolbox and flyout at left of screen.
 * @const
 */
Blockly.TOOLBOX_AT_LEFT = 2;

/**
 * ENUM for toolbox and flyout at right of screen.
 * @const
 */
Blockly.TOOLBOX_AT_RIGHT = 3;

/**
 * @enum {MyType} 
 */
var TOOLBOX = {
  TYPE_A: Blockly.TOOLBOX_AT_TOP,
  TYPE_B: Blockly.TOOLBOX_AT_BOTTOM,
  ...
};

```

I think that preserves the old values while adding type-safety by allowing us to use `enum`.

It would also be fine to record this in a TODO and address it later, because it's almost purely an annotation change.

Reply:
Added a todo to look into this more: #4248